File: Python/Count_the_Characters_V1.py
def count_characters(characters):
    count_letter = 0
    count_number = 0
    count_special = 0

    for ch in characters:
        if ch.isdigit():
            count_number += 1
        elif ch.isalpha():
            count_letter += 1
        else:
            count_special += 1

    total = count_number + count_letter + count_special
    print("Letter count:", count_letter)
    print("Number count:", count_number)
    print("Special character count:", count_special)
    print("Total number of characters:", total)


print("Count the Characters v1")
user_input = input("Enter a string: ")

if user_input == "":
    print("No input, please input a string.")
else:
    count_characters(user_input)

File: C++/Count_the_Characters_V1.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

void countCharacters(string characters) {
    int size = characters.length();
    int countLetter = 0;
    int countNumber = 0;
    int countSpecial = 0;
    int total = 0;
    char input;

    for (int counter = 0; counter < size; counter++) {
        input = characters[counter];
        if (isdigit(input)) {
            countNumber++;
        } else if (isalpha(input)) {
            countLetter++;
        } else {
            countSpecial++;
        }
    }

    total = countNumber + countLetter + countSpecial;
    cout << "Letter count: " << countLetter << endl;
    cout << "Number count: " << countNumber << endl;
    cout << "Special character count: " << countSpecial << endl;
    cout << "Total number of characters: " << total << endl;
}

int main() {
    string userInput;
    cout << "Count the Characters v1" << endl;
    cout << "Enter a string: ";
    getline(cin, userInput);

    if (userInput == "") {
        cout << "No input, please input a string." << endl;
    } else {
        countCharacters(userInput);
    }

    return 0;
}
